File: sdks/java/src/main/java/com/puxbay/resources/WebhooksResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Webhook;
import java.util.Map;

public class WebhooksResource {
    private final Puxbay client;

    public WebhooksResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Webhook> list(int page) throws PuxbayException {
        return (PaginatedResponse<Webhook>) client.request("GET", "webhooks/?page=" + page, null, PaginatedResponse.class);
    }

    public Webhook get(String webhookId) throws PuxbayException {
        return client.request("GET", "webhooks/" + webhookId + "/", null, Webhook.class);
    }

    public Webhook create(Webhook webhook) throws PuxbayException {
        return client.request("POST", "webhooks/", webhook, Webhook.class);
    }

    public Webhook update(String webhookId, Webhook webhook) throws PuxbayException {
        return client.request("PATCH", "webhooks/" + webhookId + "/", webhook, Webhook.class);
    }

    public void delete(String webhookId) throws PuxbayException {
        client.request("DELETE", "webhooks/" + webhookId + "/", null, Void.class);
    }
    
    public Map listEvents(String webhookId, int page) throws PuxbayException {
        return client.request("GET", "webhook-logs/?webhook=" + webhookId + "&page=" + page, null, Map.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/WebhooksResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Webhook
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class WebhooksResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1): PaginatedResponse<Webhook> =
        client.request(HttpMethod.Get, "webhooks/?page=$page")
    
    suspend fun get(webhookId: String): Webhook =
        client.request(HttpMethod.Get, "webhooks/$webhookId/")
    
    suspend fun create(webhook: Webhook): Webhook =
        client.request(HttpMethod.Post, "webhooks/", webhook)
    
    suspend fun update(webhookId: String, webhook: Webhook): Webhook =
        client.request(HttpMethod.Patch, "webhooks/$webhookId/", webhook)
    
    suspend fun delete(webhookId: String) {
        client.request<Unit>(HttpMethod.Delete, "webhooks/$webhookId/")
    }
    
    suspend fun listEvents(webhookId: String, page: Int = 1): Map<String, Any> =
        client.request(HttpMethod.Get, "webhook-logs/?webhook=$webhookId&page=$page")
}
